src/Metalama.Patterns.NotifyPropertyChanged/Implementation/DeferredDeclaration.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) SharpCrafters s.r.o. See the LICENSE.md file in the root directory of this repository root for details.
+
+using Metalama.Framework.Aspects;
+using Metalama.Framework.Code;
+
+namespace Metalama.Patterns.NotifyPropertyChanged.Implementation;
+
+/// <summary>
+/// A declaration that will or will not be defined later. There is no maybe.
+/// </summary>
+/// <typeparam name="T"></typeparam>
+[CompileTime]
+internal class DeferredDeclaration<T> : UncertainDeferredDeclaration<T>
```

Comment:
I think the whole concept of DeferredDeclaration (and all derived classes) could be reduced to something like this:

```cs
class Deferred<T>
{
    T _value;
    bool _isSet;
    public Value 
    {
        get 
        {
            if ( !_isSet )
                throw new InvalidOperationException();
            return _value;
        }

            set
            {
                _value = value;
                _isSet = true
            }

    }
}
```